include/deal.II/base/polynomial_space.h

```diff
@@ -170,6 +172,24 @@ class PolynomialSpace
                                    const Point<dim> &p) const;
 
   /**
+   * Computes the third derivative (grad_grad) of the <tt>i</tt>th polynomial
+   * at <tt>unit_point</tt>.
+   *
+   * Consider using compute() instead.
+   */
+  Tensor<3,dim> compute_grad_hessian (const unsigned int i,
+                                      const Point<dim> &p) const;
+
+  /**
+   * Computes the fourth derivative (grad_grad) of the <tt>i</tt>th polynomial
+   * at <tt>unit_point</tt>.
+   *
+   * Consider using compute() instead.
+   */
+  Tensor<4,dim> compute_grad_hessian_grad (const unsigned int i,
+                                           const Point<dim> &p) const;
```

Comment:
Do you actually need these functions anywhere? If not, you may consider just not having them and letting the `compute()` interface be the only way to get these higher derivatives.